flux-refineck/src/constraint_gen.rs

```diff
@@ -539,6 +551,150 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
     fn solve(self) -> Result<EVarSol, UnsolvedEvar> {
         self.evar_gen.solve()
     }
+
+    fn mk_oblig(
```

Comment:
All the functions below only require the `GlobalEnv`, let's make them all free functions taking `genv: GlobalEnv<'_, 'tcx>` as the first parameter. You'll get an error if you try to call `closure_obligs` like so

```
        let closure_obligs = closure_obligs(self.genv, did, &substs, &actuals);
```

You can get around it by copying the the global env (`let genv = self.genv`) before calling `self.infcx`.

Reply:
done.